build/helper/codegen_helper.py

```diff
@@ -134,7 +134,7 @@ def get_enum_type_check_snippet(parameter, indent):
 def _get_buffer_parameter_for_size_parameter(parameter, parameters):
     '''If parameter represents the size of another parameter in the C API, returns that other parameter. Otherwise None.'''
     for p in parameters:
-        if p['is_buffer'] and p['size']['value'] == parameter['name']:
+        if (p['is_buffer'] or p['type'] == 'ViString') and p['size']['value'] == parameter['name']:
```

Comment:
instead of `p['type'] == 'ViString'` it may be a better fit to have another piece of metadata `'is_string'` which is really closer to the intent of the if statement and clean up the code.

Reply:
I also think that `'is_buffer'` is no longer accurate. It now means `'is_array'`, right? Buffer is more about memory, and array is about the data structure.

If I understand right, then we should rename `is_buffer` (in a separate PR is fine), although `is_buffer` may still be valuable to the code-generator for some cases. `is_buffer` would be equal to `is_array or is_string`.

Reply:
Switched to using `'is_string'`

I thought about that but some are still lists and are not `array.array` or `numpy.array`. Because of that I didn't think that `'is_array'` was any better.

Reply:
I see your point. I think that in terms of metadata representation (not C or Python specific) then having something be a string makes sense. Having something be an array makes sense - this is independent of the `array.array` or `numpy.ndarray` optimizations.

The piece of metadata that would inform the generator to generate `array.array` or `numpy.ndarray` optimizations would be something like `is_waveform`.

Reply:
We need to differentiate between lists and arrays.

I may need to make them `'is_list'` and `'is_array'` when I get further along with the array PR.